src/slurm_plugin/fleet_manager.py

```diff
@@ -312,13 +319,29 @@ def _uses_single_az(self):
         subnet_ids = self._compute_resource_config.get("Networking", {}).get("SubnetIds", [])
         return len(subnet_ids) == 1
 
+    def _uses_subnet_prioritization(self):
+        return (
+            self._compute_resource_config.get("AllocationStrategy") == "prioritized"
+            and self._compute_resource_config["CapacityType"] == "on-demand"
+        ) or (
+            self._compute_resource_config.get("AllocationStrategy") == "capacity-optimized-prioritized"
+            and self._compute_resource_config["CapacityType"] == "spot"
+        )
+
     def _evaluate_launch_params(self, count):
         """Evaluate parameters to be passed to create_fleet call."""
         try:
+            enable_single_availability_zone = self._compute_resource_config.get("Networking", {}).get(
+                "SingleAvailabilityZone", None
+            )
+            if enable_single_availability_zone is None or (
+                enable_single_availability_zone and self._uses_subnet_prioritization() is False
```

Comment:
If we suppress the CLI validator and CX uses `EnableSingleVaialibilityZone` as False when using `all-or-nothing` with 1 Az and multiple Instances then Ec2 create Fleet call will Fail as we do not set SingleInstanceType or SingleAvailibilityZone which is a requirement from EC2